cmd/bootnode/main.go

```diff
@@ -108,7 +108,7 @@ func main() {
 	logConn := flag.Bool("log_conn", false, "log incoming/outgoing connections")
 	maxConnPerIP := flag.Int("max_conn_per_ip", 10, "max connections number for same ip")
 	forceReachabilityPublic := flag.Bool("force_public", false, "forcing the local node to believe it is reachable externally")
-	noTransportSecurity := flag.Bool("no_transport_security", false, "disable TLS encrypted transport")
+	noTransportSecurity := flag.Bool("no_transport_security", true, "disable TLS encrypted transport")
```

Comment:
It seems it disables transport security without proper discussion and a finalized decision from the team. Have we checked the compatibility of nodes with disabled security with the current mainnet nodes? We have to ensure they can communicate properly to avoid any network issues. Also it would be good to provide us with a few references that shows this change helps.

Reply:
Small inputs:
* Default value for the release 2024.1.0 is `"NoTransportSecurity": false`, thus `"NoTransportSecurity": true` cases is out of scope
* If mixing all possible combinations together we will end up in 16(!) possible cases on the communication between bootnode and Harmony node, but if we remove out of scope cases the number is decreasing to 9.
* the next step will be to try enable security via  `"NoTransportSecurity": false` on the bootnode 1.22 and test 3 cases

  | Bootnode1.22 Node "NoTransportSecurity": true | Bootnode1.22 Node "NoTransportSecurity": false | Bootnode1.19 Node "NoTransportSecurity": true | Bootnode1.19 Node "NoTransportSecurity": false
-- | -- | -- | -- | --
Harmony 1.19 Node "NoTransportSecurity": true |  Out of scope  | Out of scope  | Out of scope  | Out of scope 
Harmony 1.19 Node "NoTransportSecurity": false | **OK** | NOT tested  | Out of scope | **OK**
Harmony 1.22 Node "NoTransportSecurity": true | **FAILED** | NOT tested  | Out of scope  | **FAILED**
Harmony 1.22 Node "NoTransportSecurity": false | **OK** | NOT tested  | Out of scope | **OK**

Reply:
Great tests, thanks

Reply:
> the same test on the devnet have given me inverted results:

just check two devnet normal node and they had  `"NoTransportSecurity": true,`

based on test above:
1) NoTransportSecurity is not playing a role for a normal node to get the list of other node in the network (from the bootnode).
2) the non bootnode node in the network need to match their  settings `NoTransportSecurity`

Reply:
Results of canary tests all nodes have `NoTransportSecurity: false`:
* RPC node 1.22 + Bootnode 1.22 ✅
* RPC node 1.22 + Bootnode 1.19 ✅
* RPC 1.19 + Bootnode 1.22 ✅
* RPC node 1.22(95.217.207.240 aka mhe-explorers1-02) + traffic  ✅
* s1 validator 1.22  ✅
* s0 validator 1.22  ✅